Homework3/Task2/Classes/Stack.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Task2.Interfaces;
+
+namespace Task2.Classes
+{
+    public class Stack<T> : IStack<T>
+    {
+        private const int STACK_SIZE = 5;
+
+        public int Index { get; private set; }
+
+        public T[] stack = new T[STACK_SIZE];
+
+        public bool IsEmpty()
+        {
+            return Index == 0;
+        }
+
+        public T Pop()
+        {
+            return Index == 0 ? default : stack[--Index];
```

Comment:
Also, I think that --Index will decrease Index and only after that return element which is not correct.